hpx/hpx_init_impl.hpp

```diff
@@ -255,35 +343,34 @@ namespace hpx
     {
         using hpx::program_options::options_description;
 
-        options_description desc_commandline(
-            "Usage: " + app_name +  " [options]");
+        options_description desc_cmdline("Usage: " + app_name +  " [options]");
+
+        hpx::init_params iparams;
+        iparams.desc_cmdline = desc_cmdline;
+        iparams.mode = mode;
 
         if (argc == 0 || argv == nullptr)
         {
-            char *dummy_argv[2] = { const_cast<char*>(app_name.c_str()), nullptr };
-            return init(desc_commandline, 1, dummy_argv, mode);
+            return init(1, iparams.dummy_argv, iparams);
```

Comment:
Does `dummy_argv` need to be in `iparams`?

Reply:
Not really but I didn't know where to put it as a better place, cause if it static, it will live for the all program no ?

Reply:
Iether together with the other default statics or you can keep it the way it was (i.e. defined just above). If they're going to be in the struct they should at least be private and static, but I'd prefer keeping them outside.